src/api_server.py

```diff
@@ -3,36 +3,47 @@
 from dataclasses import dataclass
 from datetime import datetime
 from pathlib import Path
-from typing import Dict, List
+from typing import Dict, List, Protocol
 
 import torch
 from fastapi import FastAPI
 from loguru import logger
 from pydantic import BaseModel, validator
 
-from src.model_extractors import BenefitsAndDrawbacksExtractor
+from src.model_extractors import Predictor, QuestionAnsweringPredictor, Seq2SeqPredictor
+
+PREDICTORS: Dict[str, Predictor] = {
+    "qa": classifier_qa,
+    "seq2seq": classifier_seq2seq,
+}
 
 
 @dataclass
 class ApiServerConfig:
     prediction_log_path: str = "./data/logs/predictions/logs.jsonl"
     api_log_path: str = "./data/logs/api/logs.txt"
-    model_qa: str = "choosistant/qa-model"
-    model_seq2seq: str = "choosistant/seq2seqmodel"
+    model_name_qa: str = "choosistant/qa-model"
+    model_name_seq2seq: str = "choosistant/seq2seqmodel"
     batch_size: int = 8
 
 
 class PredictRequest(BaseModel):
     url: str
     review_text: str
-    model_type: int
+    model_type: str
 
     @validator("review_text")
     def review_text_must_not_be_empty(cls, val: str):
         if val is None or len(val.strip()) == 0:
             raise ValueError("review_text must not be empty")
         return val
 
+    @validator("model_type")
+    def model_type_(cls, val: str):
```

Comment:
The name of the method should more descriptive. Maybe something like `model_type_must_known`.